python/setup.py

```diff
@@ -35,59 +42,143 @@
+    # CMakeLists that was used to build libcuml++
+    if not os.path.exists(libcuml_path):
+        warnings.warn("Third party repositories have not been found so they  "
+                      "will be cloned. To avoid this set the environment "
+                      "variable CUML_BUILD_PATH, containing the relative "
+                      "path of the root of the repository to the folder "
+                      "where libcuml++ was built.")
+
+        clone_repo(name='cub',
+                   GIT_REPOSITORY='https://github.com/NVlabs/cub.git',
+                   GIT_TAG='v1.8.0',
+                   first=True)
+        cub_path = 'external/cub'
+
+        clone_repo(name='cutlass',
+                   GIT_REPOSITORY='https://github.com/NVIDIA/cutlass.git',
+                   GIT_TAG='v1.0.1')
+        cutlass_path = 'external/cutlass'
+
+        clone_repo(name='faiss',
+                   GIT_REPOSITORY='https://github.com/facebookresearch/faiss.git',  # noqa: E501
+                   GIT_TAG='656368b5eda4d376177a3355673d217fa95000b6')
+        # Note: faiss has a different include path
+        faiss_path = 'external'
+
+        clone_repo(name='treelite',
+                   GIT_REPOSITORY='https://github.com/dmlc/treelite.git',
+                   GIT_TAG='600afd55d1fa9bb94fc88fd3a3043cb2d5b20651')
+        treelite_path = 'external/treelite'
+
+    else:
+        treelite_path = libcuml_path + 'treelite/src/treelite'
+        faiss_path = libcuml_path + 'faiss/src/'
+        cub_path = libcuml_path + 'cub/src/cub'
+        cutlass_path = 'cutlass/src/cutlass'
 
-    cumlprims_include_dir = conda_prefix + cumlprims_include_dir
-    cumlprims_lib_dir = conda_prefix + cumlprims_lib_dir
+else:
+    subprocess.check_call(['rm', '-rf', 'external'])
+    treelite_path = ""
+    faiss_path = ""
+    cub_path = ""
+    cutlass_path = ""
 
-exc_list = []
 
-libs = ['cuda', 'cuml++', 'cumlcomms', 'nccl', 'rmm']
+##############################################################################
+# - Cython extensions build and parameters -----------------------------------
+
+libs = ['cuda',
+        'cuml++',
+        'cumlcomms',
+        'nccl',
+        'rmm']
 
 include_dirs = ['../cpp/src',
                 '../cpp/external',
                 '../cpp/src_prims',
-                '../thirdparty/cutlass',
-                '../thirdparty/cub',
-                '../thirdparty/treelite/include',
+                cutlass_path,
+                cub_path,
+                faiss_path,
+                treelite_path + '/include',
                 '../cpp/comms/std/src',
                 '../cpp/comms/std/include',
                 cuda_include_dir,
-                rmm_include_dir,
-                cumlprims_include_dir]
+                conda_include_dir]
+
+# Exclude multigpu components that use libcumlprims if --singlegpu is used
+exc_list = []
+if "--multigpu" in sys.argv:
```

Comment:
Can we use argumentparser since the args are nontrivial here?

Reply:
Good idea

Reply:
This ended up being a little more work than what I'm confortable just tacking on top of this PR (since it conflicted with `setuptools`, will create an issue for a future PR